libs/model/src/aggregates/token/GetTokenInfoAlchemy.query.ts

```diff
@@ -0,0 +1,75 @@
+import { logger, type Query } from '@hicommonwealth/core';
+import { ValidChains } from '@hicommonwealth/evm-protocols';
+import { config } from '@hicommonwealth/model';
+import * as schemas from '@hicommonwealth/schemas';
+import { Network } from 'alchemy-sdk';
+import z from 'zod';
+
+const ethChainIdToAlchemy: Record<number, Network> = {
+  [ValidChains.Base]: Network.BASE_MAINNET,
+};
+
+const errorObject = {
+  network: '',
+  address: '',
+  currency: '',
+  data: [],
+};
+
+const log = logger(import.meta);
+
+export function GetTokenInfoAlchemy(): Query<
+  typeof schemas.GetTokenInfoAlchemy
+> {
+  return {
+    ...schemas.GetTokenInfoAlchemy,
+    auth: [],
+    body: async ({ payload }) => {
+      const { eth_chain_id, token_address } = payload;
+
+      const network = ethChainIdToAlchemy[eth_chain_id];
+      if (!network) {
+        log.error(
+          `network for chain_id ${eth_chain_id} not supported by alchemy`,
+        );
+        return errorObject;
+      }
+
+      const today = new Date();
+      // Need two days ago because we need
+      // today's price + yesterday's price for 24 hour change
+      const twoDaysAgo = new Date();
+      twoDaysAgo.setHours(today.getHours() - 2 * 24);
+
+      const response = await fetch(
+        `https://api.g.alchemy.com/prices/v1/${config.ALCHEMY.APP_KEYS.PRIVATE}/tokens/historical`,
+        {
+          method: 'POST',
+          headers: {
+            'content-Type': 'application/json',
+            accept: 'application/json',
+          },
+          body: JSON.stringify({
+            startTime: twoDaysAgo.toISOString(),
+            endTime: today.toISOString(),
+            interval: '1d',
+            withMarketData: true,
+            network,
+            address: token_address,
+          }),
+        },
+      );
+
+      const json = await response.json();
+
+      if (json?.error) {
+        log.error(
+          json.error.message ||
+            'Unknown error from alchemy in GetTokenInfoAlchemy query',
+        );
+        return errorObject;
+      }
+      return json as z.infer<typeof schemas.GetTokenInfoAlchemy.output>;
```

Comment:
Y not parse the json and then return it? Otherwise, if the schema doesn't match in the future could lead to some frustrating errors.